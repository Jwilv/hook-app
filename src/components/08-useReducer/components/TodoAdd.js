import React from 'react'
import { useForm } from '../../../hooks/useForm'

export const TodoAdd = ({handleAddTodo}) => {

    const [{ description }, handleInputChanGet, reset] = useForm({
        description: ''
    })

    const handleSubmit = (event) => {
        event.preventDefault();

        if (description.trim().length <= 1) return;

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false,
        }

        handleAddTodo(newTodo);
        reset();
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type='text'
                name='description'
                placeholder='Incluir...'
                autoComplete='off'
                className='form-control'
                value={description}
                onChange={handleInputChanGet}
            />
            <button
                type='submit'
                className='btn btn-block btn-outline-primary mt-1 btn-block'
            >
                Agregar
            </button>
        </form>
    )
}
