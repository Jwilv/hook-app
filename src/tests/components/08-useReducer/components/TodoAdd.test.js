import React from 'react'
import { shallow } from 'enzyme'
import { TodoAdd } from '../../../../components/08-useReducer/components/TodoAdd';


describe('pruebas en <TodoAdd />', () => { 

    const handleAddTodo = jest.fn();

    const wrapper = shallow( <TodoAdd 
        handleAddTodo={handleAddTodo}/> );

    test('debe de mostrarse corectamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('NO debe de llamar a handleAddTodo', () => {
        //una manera nueva de simular el evento
        const formSubmit = wrapper.find('form').prop('onSubmit')
        formSubmit( {preventDefault(){} } );

        expect( handleAddTodo ).toHaveBeenCalledTimes(0);
    });

    test('debe de llamar a la funcion handleAddTodo', () => {
        const value = 'si se pudo'
        wrapper.find('input').simulate('change',{
            target:{
                value,
                name : 'description'
            }
        })

        const formSubmit = wrapper.find('form').prop('onSubmit')
        formSubmit( {preventDefault(){} } );

        expect( handleAddTodo ).toHaveBeenCalledTimes(1);
        expect( handleAddTodo ).toHaveBeenCalledWith( expect.any(Object));
        expect( handleAddTodo ).toHaveBeenCalledWith( {
            id: expect.any(Number),
            desc: value,
            done: false,
        });

        //luego del submit se debe ejecutar el reset
        expect( wrapper.find('input').prop('value')).toBe('')
    });
    
 })