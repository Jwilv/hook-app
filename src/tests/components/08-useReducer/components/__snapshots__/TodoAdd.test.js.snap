// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TodoAdd /> debe de mostrarse corectamente  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    autoComplete="off"
    className="form-control"
    name="description"
    onChange={[Function]}
    placeholder="Incluir..."
    type="text"
    value=""
  />
  <button
    className="btn btn-block btn-outline-primary mt-1 btn-block"
    type="submit"
  >
    Agregar
  </button>
</form>
`;
